Make feedback component more responsive and usable on mobile

vanitabarrett opened the pull request:
## What
- Implement new design for feedback component, with larger touch targets and more responsive layout.
- Tidy up unused feedback component code
- Remove outdated sections of feedback component documentation

<!-- Remember to add this to the CHANGELOG if applicable -->

## Why
Small touch targets and small spacing between items makes it visually frustrating, but for mobile users this might make it harder to click on the intended link.
The previous design also failed when a user zoomed, as the links started to overflow the container.

## Before and After (desktop)
<img width="983" alt="Screenshot 2020-03-09 at 10 42 08" src="https://user-images.githubusercontent.com/29889908/76205763-cdbe4480-61f2-11ea-9649-525eb982eb6b.png">

<img width="990" alt="Screenshot 2020-03-09 at 10 41 59" src="https://user-images.githubusercontent.com/29889908/76205934-137b0d00-61f3-11ea-8cba-23cbd1beac0d.png">

## Before and After (mobile)
<img width="317" alt="Screenshot 2020-03-09 at 10 42 35" src="https://user-images.githubusercontent.com/29889908/76205952-1ece3880-61f3-11ea-9966-3c2eb49e8a08.png">
<img width="435" alt="Screenshot 2020-03-09 at 10 42 50" src="https://user-images.githubusercontent.com/29889908/76205953-1f66cf00-61f3-11ea-9c68-02a6e02c69d7.png">

kr8n3r commented:
nice one for splitting the questions and being able to remove the grid 👍

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -35,39 +39,87 @@
 }
 
+.js-enabled .gem-c-feedback__prompt-questions {
+  display: inline-block;
```
Why is this set? It seems to be overriding a `display: block`, removing it doesn't seem to make any difference?

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -35,39 +39,87 @@
+  border-bottom: 1px solid govuk-colour("white");
+  padding: govuk-spacing(5) govuk-spacing(5) govuk-spacing(3) govuk-spacing(5);
+  box-sizing: border-box;
+  width: 100%;
```
This width and the position relative on line 47 don't appear to do anything, but I could be wrong.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -186,79 +238,30 @@
   }
 
   .gem-c-feedback__option-list-item {
+    position: relative;
```
Am I right in thinking this is to capture the pseudo element on the link? Only the link itself already has position relative, so I'm not sure this is needed?

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -186,79 +238,30 @@
 
   .gem-c-feedback__option-list-item {
+    position: relative;
+    display: inline-block;
```
It's quite hard to follow the inheritance tree through all of this (due to the current duplication of the CSS, not this PR) but I think there's some simplification to be had here. The heading `gem-c-feedback__prompt-question` is also floated and has margin to position it, but I think if all of the floating was removed and everything set to inline-block the required layout can be achieved with fewer lines of code.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -1,19 +1,16 @@
+  }
+
+  .visually-hidden {
+    @include govuk-visually-hidden;
```
This isn't your change (in fact it's probably mine! but I can't remember) but do you know why the include is here, rather than using the class directly in the component? Seems like something worth fixing.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -1,19 +1,16 @@
-
   @include govuk-media-query($from: tablet) {
     margin-top: govuk-spacing(9);
+    border-bottom: 1px solid govuk-colour("white");
```
Suggestion: currently this swaps between the `.gem-c-feedback` element having the border bottom above tablet and the `.gem-c-feedback__prompt-questions` having it on mobile (where both need it to have two, when stacked) - would it be possible to just have the border bottom on the `.gem-c-feedback__prompt-questions` elements and not need any media queries involved?

andysellick replied:
Actually, I've just realised why this is difficult - at a certain breakpoint the heights of the two elements are inconsistent. Hmm. Maybe don't worry about this. Although maybe there's a solution..

![Screenshot 2020-03-11 at 13 47 42](https://user-images.githubusercontent.com/861310/76423630-ff740e80-639e-11ea-9807-bfdedd6b9f2e.png)

vanitabarrett replied:
Applying the border to `.gem-c-feedback__prompt-questions` was my first approach, but I also noticed the issue you had above so switched approaches. I tried again, but I think either way will require media queries.

andysellick replied:
Yeah, fair enough.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_feedback.scss:
```diff
@@ -80,6 +117,30 @@
 }
 
+.js-enabled {
+  .gem-c-feedback__option-list {
```
Just wondering why there's so many rules nested beneath `.js-enabled` - doesn't the component simply not appear if JS is disabled?

vanitabarrett replied:
Yes, good point, I don't think this is needed any more since I tidied up some of the CSS. Will remove

vanitabarrett commented:
@andysellick Thanks for your review! I've addressed all your comments now, apart from the one about the border - I tried to find an alternative solution but couldn't find anything that would be simpler.

injms commented:
The link "Don't have an email address?" is the same text size as "Yes" and "No", but doesn't have the pseudo element to increase the touch target. Should the pseudo element be added to the link to also give it a larger touch target?

vanitabarrett commented:
Thanks @injms , good spot! I've increased the touch target on that link now

alex-ju commented:
🙌 for using pseudo-elements on this
